### PR TITLE
[types] Remove unnecessary deps

### PR DESCRIPTION
**Pull request by c410-f3r**

### CODE REVIEW

File: types/src/v2/error.rs [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
-	#[error("invalid params")]
 	InvalidParams,
 }
 
+impl fmt::Debug for RpcError {
+	#[inline]
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		match *self {
+			Self::Unknown => write!(f, "unknown rpc error"),
+			Self::InvalidParams => write!(f, "invalid paramsr"),
```
Comment (niklasad1) [inline types/src/v2/error.rs:16]:
```suggestion
			Self::InvalidParams => write!(f, "invalid params"),
```

**Review by niklasad1 — COMMENTED** [review]
Thanks for your PR.

Looks overall good but I don't really understand the rationale behind making this crate `no-std` without the client traits then it's not super useful, such as the proc macros wouldn't work. I suppose making the `SubscriptionClient trait` std only (subscription type) and make `Client trait` no-std would do it.

Can you elaborate why making the `no-std` is useful?

**Comment by c410-f3r** [comment]
`no_std` support is usually a complementary and natural step when removing unnecessary dependencies. Besides, isn't it something desirable in the long run?

If necessary, I can remove `no_std` stuff. No problem.

**Comment by niklasad1** [comment]
> no_std support is usually a complementary and natural step when removing unnecessary dependencies. Besides, isn't it something desirable in the long run?

Sure, why not it introduces some boiler plate code for the `error stuff` but worth it if somebody wants to use the types in `no-std` but we have no plans making client/server to support for `no-std` but we have plans to move the [HTTP client to use surf as backend for WASM support](https://github.com/paritytech/jsonrpsee/issues/227)

**Review by niklasad1 — CHANGES_REQUESTED** [review]
LGTM except the typo, see inline suggestion.

**Review by niklasad1 — APPROVED** [review]

**Comment by c410-f3r** [comment]
IIRC, `WASM` language isn't aware of `std` stuff (filesystem, threads, timer, etc) and that is one of the reasons for the creation of `WASI`.

**Comment by niklasad1** [comment]
> IIRC, WASM language isn't aware of std stuff (filesystem, threads, timer, etc) and that is one of the reasons for the creation of WASI.

Yes, but it's possible to do with `JavaScript bindings` too as [surf/http-client](https://github.com/http-rs/http-client) does.

**Comment by c410-f3r** [comment]
Cool! Thanks for the link

File: types/src/error.rs [hunk 2]
```diff
@@ -16,62 +18,110 @@ impl<T: fmt::Display> fmt::Display for Mismatch<T> {
...
 	/// Configured max number of request slots exceeded.
-	#[error("Configured max number of request slots exceeded")]
 	MaxSlotsExceeded,
 	/// Custom error.
-	#[error("Custom error: {0}")]
 	Custom(String),
 }
 
+impl fmt::Debug for Error {
+	#[inline]
```
Comment (dvdplm) [inline types/src/error.rs:56]: Let's not try to outsmart the compiler; formatting errors is not on the hot path for anything so please remove the inlining hints.

File: types/src/error.rs [hunk 3]
```diff
@@ -16,62 +18,110 @@ impl<T: fmt::Display> fmt::Display for Mismatch<T> {
...
+			}
+			Self::WsRequestTimeout => write!(f, "Websocket request timeout"),
+			Self::MaxSlotsExceeded => write!(f, "Configured max number of request slots exceeded"),
+			Self::Custom(ref elem) => write!(f, "Custom error: {}", elem),
+		}
+	}
+}
+
+impl fmt::Display for Error {
+	#[inline]
```
Comment (dvdplm) [inline types/src/error.rs:86]: same

**Review by dvdplm — CHANGES_REQUESTED** [review]
Reducing compile times is great, thank you so much for looking into this.

I agree with @niklasad1 that `no-std` is not a goal and adds more noise than it's worth imo. I also think that the added boilerplate around errors is questionable. I prefer it the way it was.

Any chance we can keep the compile time improvements without the inlining, no-std, and error re-implementations?

File: types/src/v2/traits.rs [hunk 4]
```diff
@@ -11,5 +11,7 @@ impl<R, T> RpcMethod<R> for T where T: Fn(RpcParams) -> Result<R, RpcError> + Se
 
 /// RPC Call Result.
 trait RpcResult {
```
Comment (niklasad1) [inline types/src/v2/traits.rs:13]: please merge with master, this was removed recently.

**Comment by niklasad1** [comment]
@dvdplm please take another look at this PR ^^

**Review by dvdplm — APPROVED** [review]

**Pull request merged**